pkg/querier/querier_test.go

```diff
@@ -228,6 +228,93 @@ func TestNoHistoricalQueryToIngester(t *testing.T) {
 
 }
 
+func TestNoFutureQueries(t *testing.T) {
+	testCases := []struct {
+		name               string
+		mint, maxt         time.Time
+		hitStores          bool
+		maxQueryIntoFuture time.Duration
+	}{
+		{
+			name:               "hit-test1",
+			mint:               time.Now().Add(-5 * time.Hour),
+			maxt:               time.Now().Add(1 * time.Hour),
+			hitStores:          true,
+			maxQueryIntoFuture: 10 * time.Minute,
+		},
+		{
+			name:               "hit-test2",
+			mint:               time.Now().Add(-5 * time.Hour),
+			maxt:               time.Now().Add(-59 * time.Minute),
+			hitStores:          true,
+			maxQueryIntoFuture: 10 * time.Minute,
+		},
+		{ // Skipping stores is disabled.
+			name:               "hit-test2",
+			mint:               time.Now().Add(-5 * time.Hour),
+			maxt:               time.Now().Add(-50 * time.Minute),
+			hitStores:          true,
+			maxQueryIntoFuture: 0,
```

Comment:
In this test you would hit the store even if `maxQueryIntoFuture` is enabled, because the `mint` and `maxt` are in the past. You should set them in the future to assert that when `maxQueryIntoFuture` is disabled the stores will be hit anyway.